### PR TITLE
run preflight checks after mode switch has changed

### PR DESCRIPTION
**Pull request by tumbili**
This addresses #3054

### CODE REVIEW

**Comment by tumbili** [comment]
@LorenzMeier The problem was that condition_sensors_intialised was never true because the main mode switch was assigned after the rc calibration.

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -1272,6 +1274,7 @@ int commander_thread_main(int argc, char *argv[])
 	int autosave_params; /**< Autosave of parameters enabled/disabled, loaded from parameter */
 
 	int32_t disarm_when_landed = 0;
+	int32_t map_mode_sw;
```
Comment (LorenzMeier) [inline src/modules/commander/commander.cpp]: Dangling value. This has to be initialized to something.

**Comment by LorenzMeier** [comment]
Nice. Will merge this once it passed.

**Comment by tumbili** [comment]
Merged

**Pull request merged**